cibyl/sources/elasticsearch/api.py

```diff
@@ -355,6 +365,116 @@ def get_deployment(self, **kwargs):
+one')
+        job_builds_found = self.get_builds(**kwargs)
+        query_body = {
+            "query": {
+                "bool": {
+                    "must": [
+                        {},
+                        {
+                            "bool": {
+                              "should": [
+                                {}
+                              ]
+                            }
+                        },
+                        {
+                            "exists": {
+                                "field": "test_status"
+                            }
+                        },
+                        {
+                            "exists": {
+                                "field": "test_name"
+                            }
+                        }
+                    ]
+                }
+            },
+            "sort": [{"timestamp.keyword": {"order": "desc"}}]
+        }
+
+        hits = []
+        for job in job_builds_found:
+            query_body['query']['bool']['must'][0] = {
+                "match": {
+                    "job_name.keyword": f"{job}"
+                }
+            }
+            builds = job_builds_found[job].builds
+            for build_number in builds:
+                query_body['query']['bool']['must'][1]['bool']['should'] = {
+                    "match": {
+                        "build_num": build_number
+                    }
+                }
+                results = self.__query_get_hits(
+                    query=query_body,
+                    index='logstash_jenkins'
+                )
+
+                # We need to process all the results because
+                # in the same build we can have different tests
+                # So we can't use '"size": 1' in the query
+                if results:
+                    for result in results:
+                        hits.append(result)
+
+        for hit in hits:
+            job_name = hit['_source']['job_name']
+            build_number = hit['_source']['build_num']
+            test_name = hit['_source']['test_name']
+            test_status = hit['_source']['test_status']
+            class_name = hit['_source'].get(
+                'test_class_name',
+                None
+            )
+            # Some build is not parsed good and contains
+            # More info than a time in the field
+            try:
+                test_duration = hit['_source'].get(
+                    'test_time',
+                    None
+                )
+                if test_duration:
+                    test_duration = float(test_duration)*1000
+            except ValueError:
+                LOG.debug("'test_time' field is not well parsed in "
+                          "elasticsearch for job: %s and build ID: %s",
+                          job_name,
+                          build_number
+                          )
+                continue
+
+            build_object = Build(
```

Comment:
Do you need to create this Build object? The call to get_builds should ensure that is already created, shouldn't it? If so, you could skip lines 462-466 and just add the test

Reply:
Since I changed the call from `get_jobs` to `get_builds` and we don't have calls to Elasticsearch to get the builds from this method, no, we don't need this build object. Thx. Change done.